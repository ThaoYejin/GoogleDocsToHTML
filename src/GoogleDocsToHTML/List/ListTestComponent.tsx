import { docs_v1 } from "googleapis";
import React from "react";
import { List } from "./List";

export interface ListTestComponentProps {
    doc: docs_v1.Schema$Document, 
    startIndex: number
}

export const ListTestComponent: React.FC<ListTestComponentProps> = ({
   doc,
   startIndex
}
) => {
    var content = List(doc, startIndex);
    var listLength = content[1];
    return (
        <>
            {listLength}
            {content[0]}
        </>
    )
}