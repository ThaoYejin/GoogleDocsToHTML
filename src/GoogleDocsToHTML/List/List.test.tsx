
import React from 'react';
import renderer from 'react-test-renderer';
import { ListTestComponent } from "./ListTestComponent";
import * as doc1 from './OrderedList.json';
import * as doc2 from './UnorderedList.json';

it("ol runs correctly", ()=>{
    var startIndex = 1;
    const tree = renderer
        .create(<ListTestComponent doc={doc1} startIndex={startIndex} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})

it("ul runs correctly", ()=>{
    var startIndex = 1;
    const tree = renderer
        .create(<ListTestComponent doc={doc2} startIndex={startIndex} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
})