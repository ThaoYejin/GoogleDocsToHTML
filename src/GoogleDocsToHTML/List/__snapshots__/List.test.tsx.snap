// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ol runs correctly 1`] = `
Array [
  "3",
  <ol>
    <li>
      <span
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        Item 1

      </span>
    </li>
    <li>
      <span
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        Item 2

      </span>
    </li>
    <li>
      <span
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        Item 3
      </span>
    </li>
  </ol>,
]
`;

exports[`ul runs correctly 1`] = `
Array [
  "3",
  <ul>
    <li>
      <span
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        Item 1

      </span>
    </li>
    <li>
      <span
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        Item 2

      </span>
    </li>
    <li>
      <span
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        Item 3
      </span>
    </li>
  </ul>,
]
`;
